### PR TITLE
fix #784

### LINKED ISSUE #784: Improve Frequency
## 🚀 Feature

If we would like to log datapoints/second every 100 iterations, we most probably do like this 
```python
wps_metric = Frequency(output_transformer=lambda x: x['ntokens'])
wps_metric.attach(trainer, name='wps', event_name=Events.ITERATION_COMPLETED(every=100))
```
however, seems like this wont take into account all other iterations while computing the total number of tokens.
```python
class Frequency(Metric):
    ....
    def attach(self, engine, name, event_name=Events.ITERATION_COMPLETED):
        engine.add_event_handler(Events.EPOCH_STARTED, self.started)
        engine.add_event_handler(event_name, self.iteration_completed)
        engine.add_event_handler(event_name, self.completed, name)
```
IMO, should be 
```python
class Frequency(Metric):
    ....
    def attach(self, engine, name, event_name=Events.ITERATION_COMPLETED):
        engine.add_event_handler(Events.EPOCH_STARTED, self.started)
        engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_completed)
        engine.add_event_handler(event_name, self.completed, name)
```

cc @erip

### PR DESCRIPTION
**Pull request by erip**
Fixes #784

Description: Explicitly wires `self.iteration_completed` as the handler for `Events.ITERATION_COMPLETED` instead of the user-provided one, which handles edge cases surrounding `Events`-specific kwargs.

~TODO: write some good tests. 😄~

Check list:
* [x] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by erip** [comment]
Hopefully I understood the issue correctly - the tests seem to be OK. If these tests look representative, I think they can (and should) be refactored to generalize the original frequency tests w/ the full engine setup.

File: tests/ignite/metrics/test_frequency.py [hunk 1]
```diff
@@ -57,3 +57,45 @@ def test_frequency_with_engine():
...
+
+    estimated_wps = batch_size * workers / artificial_time
+
+    def update_fn(engine, batch):
+        time.sleep(artificial_time)
+        return {"ntokens": len(batch)}
+
+    engine = Engine(update_fn)
+    wps_metric = Frequency(output_transform=lambda x: x["ntokens"], device=device)
+    wps_metric.attach(engine, "wps")
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_frequency.py:76]: Shouldn't this be `wps_metric.attach(engine, "wps", event_name=Events.ITERATION_COMPLETED(every=every))` ?

**Comment by vfdev-5** [comment]
Yes, you are right, we can refactor original and new tests into a single code. 

And there is a docstring update.

**Comment by erip** [comment]
This looks good now. 😄 Which docstring should be updated? Just adding another example in the `Frequency` docstring?

File: tests/ignite/metrics/test_frequency.py [hunk 2]
```diff
@@ -35,9 +35,10 @@ def update_fn(engine, batch):
 
     engine = Engine(update_fn)
     wps_metric = Frequency(output_transform=lambda x: x["ntokens"], device=device)
-    wps_metric.attach(engine, "wps")
+    event = Events.ITERATION_COMPLETED if every == 1 else Events.ITERATION_COMPLETED(every=every)
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_frequency.py:38]: Maybe we can just use `Events.ITERATION_COMPLETED(every=every)` ? It works now with `every=1`.

Comment (erip) [reply tests/ignite/metrics/test_frequency.py:38]: Is it the case that `Events.ITERATION_COMPLETED(every=1)` is equivalent to `Events.ITERATION_COMPLETED`? For some reason I thought there was some nuanced difference... maybe not, though.

**Comment by vfdev-5** [comment]
> Just adding another example in the Frequency docstring?

Yes, another example with the explanation of what it does :)

File: tests/ignite/metrics/test_frequency.py (hunk 2 above)
Comment (erip) [reply tests/ignite/metrics/test_frequency.py:38]: Ah - just checked the `CallableEventWithFilter.__call__`; it seems like they're forced to equivalent explicitly. Will update the test.

Comment (vfdev-5) [reply tests/ignite/metrics/test_frequency.py:38]: Now `Events.ITERATION_COMPLETED(every=1) == Events.ITERATION_COMPLETED`.
https://github.com/pytorch/ignite/blob/master/ignite/engine/events.py#L78

**Comment by vfdev-5** [comment]
Elijah, I was also wondering how's going integrations of training examples of fairseq with ignite you were working on previously ?

**Comment by erip** [comment]
It's going slowly, unfortunately. Fairseq has been working on moving from a framework to a library and it's made development outside of adding new architecture slow. I have a handful of pending PRs that would hopefully move the needle a bit, but there hasn't been much movement.

**Comment by vfdev-5** [comment]
I see, thanks for the info. Yes, maybe also they do not have a point to use such libs like ignite (and introduce it as a dependency)...

Are you at PyTorch slack where we could discuss in more fluent way ?

**Comment by erip** [comment]
@vfdev-5 I just requested access. :-)

**Comment by vfdev-5** [comment]
> @vfdev-5 I just requested access. :-)

Sounds good! If you have no answer from them within 1-2 days, please give me your email, I can try to make a request directly from the application (and it can be accepted more quickly :)
